core/src/test/java/org/elasticsearch/threadpool/ScalingThreadPoolTests.java

```diff
@@ -177,7 +177,8 @@ public void testScalingThreadPoolThreadsAreTerminatedAfterKeepAlive() throws Int
                     }
                 });
             }
-            assertThat(stats(threadPool, threadPoolName).getThreads(), equalTo(128));
+            int threads = stats(threadPool, threadPoolName).getThreads();
+            assertEquals(128, threads);
```

Comment:
Yeah I tried to reproduce with an isolated test, going as far as trying to replicate the exact class hierarchy/generics/signatures, but I could never get it to fail...